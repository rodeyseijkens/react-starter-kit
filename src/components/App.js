import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Provider as ReduxProvider } from 'react-redux';

const ContextType = {
  // Enables critical path CSS rendering
  // https://github.com/kriasoft/isomorphic-style-loader
  insertCss: PropTypes.func.isRequired,
  // Universal HTTP client
  fetch: PropTypes.func.isRequired,
  pathname: PropTypes.string.isRequired,
  query: PropTypes.shape(),
  // Integrate Redux
  // http://redux.js.org/docs/basics/UsageWithReact.html
  ...ReduxProvider.childContextTypes,
};

/**
 * The top-level React component setting context (global) variables
 * that can be accessed from all the child components.
 *
 * https://facebook.github.io/react/docs/context.html
 *
 * Usage example:
 *
 *   const context = {
 *     history: createBrowserHistory(),
 *     store: createStore(),
 *   };
 *
 *   ReactDOM.render(
 *     <App context={context}>
 *       <Layout>
 *         <LandingPage />
 *       </Layout>
 *     </App>,
 *     container,
 *   );
 */
class App extends PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };

  static childContextTypes = ContextType;

  getChildContext() {
    const { context } = this.props;
    return context;
  }

  render() {
    const { children } = this.props;
    // NOTE: If you need to add or modify header, footer etc. of the app,
    // please do that inside the Layout component.
    return React.Children.only(children);
  }
}

export default App;
