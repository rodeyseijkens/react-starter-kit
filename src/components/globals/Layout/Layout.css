@import '../variables.css';

/*
 * Base styles
 * ========================================================================== */
*,
*::before,
*::after {
  box-sizing: border-box;
}

html {
  color: #222;
  font-family: var(--default-font-family);
  font-size: var(--default-font-size); /* ~16px; */
  font-weight: var(--default-font-weight-regular);
  line-height: var(--default-line-height); /* ~22px */
  -ms-overflow-style: scrollbar;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
}

body {
  margin: 0;
}

:global(#app) {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}

/* Suppress the focus outline on elements that cannot be accessed via keyboard.
 * This prevents an unwanted focus outline from appearing around elements that
 * might still respond to pointer events.
 *
 * Credit: https://github.com/suitcss/base
 */
[tabindex='-1']:focus {
  outline: 0 !important;
}

/*
 * Remove text-shadow in selection highlight:
 * https://twitter.com/miketaylr/status/12228805301
 *
 * These selection rule sets have to be separate.
 * Customize the background color to match your design.
 */

::-moz-selection,
::selection {
  background: color(var(--color-primary) alpha(75%));
  text-shadow: none;
}

/*
 * A better looking default horizontal rule
 */

hr {
  border: 0;
  border-top: 1px solid #ccc;
  box-sizing: content-box;
  display: block;
  height: 1px;
  margin: 1em 0;
  padding: 0;
}

/* Reset margins on paragraphs
 * Similarly, the top margin on `<p>`s get reset. However, we also reset the
 * bottom margin to use `rem` units instead of `em`.
 */
p {
  line-height: 1.5;
  margin-bottom: var(--default-paragraph-margin-bottom);
  margin-top: 0;
}

/* Abbreviations */

abbr[title],
abbr[data-original-title] {
  border-bottom: 0;
  cursor: help;
  text-decoration: underline;
  text-decoration: underline dotted;
}

address {
  font-style: normal;
  line-height: inherit;
  margin-bottom: 1rem;
}

ol,
ul,
dl {
  margin-bottom: 1rem;
  margin-top: 0;
}

ol ol,
ul ul,
ol ul,
ul ol {
  margin-bottom: 0;
}

dt {
  font-weight: var(--default-font-weight-medium);
}

dd {
  margin-bottom: 0.5rem;
  margin-left: 0;
}

blockquote {
  margin: 0 0 1rem;
}

dfn {
  font-style: italic; /* Add the correct font style in Android 4.3- */
}

b,
strong {
  font-weight: bolder; /* Add the correct font weight in Chrome, Edge, and Safari */
}

small {
  font-size: 80%; /* Add the correct font size in all browsers */
}

/*
 * Prevent `sub` and `sup` elements from affecting the line height in
 * all browsers.
 */

sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}

sub {
  bottom: -0.25em;
}

sup {
  top: -0.5em;
}

/*
 * Links
 */

a {
  background-color: transparent; /* Remove the gray background on active links in IE 10. */
  color: var(--default-link-color);
  text-decoration: var(--default-link-decoration);
  -webkit-text-decoration-skip: objects; /* Remove gaps in links underline in iOS 8+ and Safari 8+. */

  &:hover {
    color: var(--default-link-hover-color);
  }
}

/*
 * Remove the gap between audio, canvas, iframes,
 * images, videos and the bottom of their containers:
 * https://github.com/h5bp/html5-boilerplate/issues/440
 */

audio,
canvas,
iframe,
img,
svg,
video {
  vertical-align: middle;
}

/*
 * Remove default fieldset styles.
 */

fieldset {
  border: 0;
  margin: 0;
  min-width: 0;
  padding: 0;
}

/*
 * Allow only vertical resizing of textareas.
 */

textarea {
  resize: vertical;
}

/*
 * Browser upgrade prompt
 * ========================================================================== */

:global(.browserupgrade) {
  background: #ccc;
  color: #000;
  margin: 0.2em 0;
  padding: 0.2em 0;
}

/*
 * Print styles
 * Inlined to avoid the additional HTTP request:
 * http://www.phpied.com/delay-loading-your-print-css/
 * ========================================================================== */

@media print {
  *,
  *::before,
  *::after {
    background: transparent !important;
    box-shadow: none !important;
    color: #000 !important; /* Black prints faster: http://www.sanbeiji.com/archives/953 */
    text-shadow: none !important;
  }

  a,
  a:visited {
    text-decoration: underline;
  }

  a[href]::after {
    content: ' (' attr(href) ')';
  }

  abbr[title]::after {
    content: ' (' attr(title) ')';
  }

  /*
   * Don't show links that are fragment identifiers,
   * or use the `javascript:` pseudo protocol
   */

  a[href^='#']::after,
  a[href^='javascript:']::after {
    content: '';
  }

  pre,
  blockquote {
    border: 1px solid #999;
    page-break-inside: avoid;
  }

  /*
   * Printing Tables:
   * http://css-discuss.incutio.com/wiki/Printing_Tables
   */

  thead {
    display: table-header-group;
  }

  tr,
  img {
    page-break-inside: avoid;
  }

  img {
    max-width: 100% !important;
  }

  p,
  h2,
  h3 {
    orphans: 3;
    widows: 3;
  }

  h2,
  h3 {
    page-break-after: avoid;
  }
}
