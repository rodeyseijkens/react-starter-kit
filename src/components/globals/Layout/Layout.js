import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';

import themableWithStyles from 'themableWithStyles';

import styles from './Layout.css';

import Header from '../../views/Header';
import Footer from '../../views/Footer';
import Feedback from '../../views/Feedback';

@themableWithStyles(styles)
class Layout extends PureComponent {
  static propTypes = {
    contentOnly: PropTypes.bool,
    children: PropTypes.node.isRequired,
  };

  static defaultProps = {
    contentOnly: false,
  };

  render() {
    const { contentOnly, children } = this.props;

    if (contentOnly) return children;

    return (
      <Fragment>
        <Header />
        {children}
        <Feedback />
        <Footer />
      </Fragment>
    );
  }
}

export default Layout;
