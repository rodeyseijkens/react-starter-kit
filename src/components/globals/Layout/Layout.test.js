/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';

import App from '../../App';
import Layout from './Layout';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};

describe('<Layout />', () => {
  test('renders children correctly', () => {
    const store = mockStore(initialState);
    const wrapper = mount(
      <App
        context={{
          pathname: '',
          insertCss: () => {},
          fetch: () => {},
          store,
        }}
      >
        <Layout>
          <div className="child" />
        </Layout>
      </App>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
