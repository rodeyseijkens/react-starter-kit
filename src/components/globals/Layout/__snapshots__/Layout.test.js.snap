// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders children correctly 1`] = `
<App
  context={
    Object {
      "fetch": [Function],
      "insertCss": [Function],
      "pathname": "",
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    }
  }
>
  <ThemableWithStyles(Layout)
    contentOnly={false}
    css={Object {}}
  >
    <Layout
      contentOnly={false}
      css={Object {}}
    >
      <ThemableWithStyles(Header)
        css={Object {}}
      >
        <Header
          css={Object {}}
        >
          <header
            className="root"
          >
            <div
              className="container"
            >
              <ThemableWithStyles(Navigation)
                css={Object {}}
              >
                <Navigation
                  css={Object {}}
                >
                  <nav
                    className="root"
                    role="navigation"
                  >
                    <Link
                      className="link"
                      onClick={null}
                      to="/about"
                    >
                      <a
                        className="link"
                        href="/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                    </Link>
                    <Link
                      className="link"
                      onClick={null}
                      to="/contact"
                    >
                      <a
                        className="link"
                        href="/contact"
                        onClick={[Function]}
                      >
                        Contact
                      </a>
                    </Link>
                  </nav>
                </Navigation>
              </ThemableWithStyles(Navigation)>
              <Link
                className="brand"
                onClick={null}
                to="/"
              >
                <a
                  className="brand"
                  href="/"
                  onClick={[Function]}
                >
                  <img
                    alt="React"
                    height="38"
                    src="logo-small.png"
                    srcSet="logo-small@2x.png 2x"
                    width="38"
                  />
                  <span
                    className="brandTxt"
                  >
                    Your Company
                  </span>
                </a>
              </Link>
              <div
                className="banner"
              >
                <h1
                  className="bannerTitle"
                >
                  React
                </h1>
                <p
                  className="bannerDesc"
                >
                  Complex web apps made easy
                </p>
              </div>
            </div>
          </header>
        </Header>
      </ThemableWithStyles(Header)>
      <div
        className="child"
      />
      <ThemableWithStyles(Feedback)
        css={Object {}}
      >
        <Feedback
          css={Object {}}
        >
          <section
            className="root"
          >
            <div
              className="container"
            >
              <a
                className="link"
                href="https://gitter.im/kriasoft/react-starter-kit"
                rel="noopener noreferrer"
                target="_blank"
              >
                Ask a question
              </a>
              <span
                className="spacer"
              >
                |
              </span>
              <a
                className="link"
                href="https://github.com/kriasoft/react-starter-kit/issues/new"
                rel="noopener noreferrer"
                target="_blank"
              >
                Report an issue
              </a>
            </div>
          </section>
        </Feedback>
      </ThemableWithStyles(Feedback)>
      <ThemableWithStyles(Footer)
        css={Object {}}
      >
        <Footer
          css={Object {}}
        >
          <footer
            className="root"
          >
            <div
              className="container"
            >
              <span
                className="text"
              >
                © Your Company
              </span>
              <span
                className="spacer"
              >
                ·
              </span>
              <Link
                className="link"
                onClick={null}
                to="/"
              >
                <a
                  className="link"
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </Link>
              <span
                className="spacer"
              >
                ·
              </span>
              <Link
                className="link"
                onClick={null}
                to="/privacy"
              >
                <a
                  className="link"
                  href="/privacy"
                  onClick={[Function]}
                >
                  Privacy
                </a>
              </Link>
              <span
                className="spacer"
              >
                ·
              </span>
              <Link
                className="link"
                onClick={null}
                to="/not-found"
              >
                <a
                  className="link"
                  href="/not-found"
                  onClick={[Function]}
                >
                  Not Found
                </a>
              </Link>
            </div>
          </footer>
        </Footer>
      </ThemableWithStyles(Footer)>
    </Layout>
  </ThemableWithStyles(Layout)>
</App>
`;
